### PR TITLE
Revision peatland module v2 realisation + bugfix output/extra/disaggregation.R related to 67k 

### PR DESCRIPTION
**Pull request by flohump**
### changed
- **default.cfg** Default for cfg$gms$peatland changed from `on` to `v2`

### removed
- **58_peatland** removed realization "on"

Changes in peatland module `v2` realization:
- Improved initialization of managed peatland
- Correction of scaling factor
- Emission factors for intact peatland are now considered in the internal GHG emission pricing. The magpie4 library has been adjusted accordingly. 
- **scripts** added peatland to output/extra/disaggregation.R

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

Current develop: `PR611_SSP2-Ref-PeatOn`
This PR: `PR611Final4_SSP2-Ref-PeatV2`

Resources_Land_Cover_Cropland-91
Productivity_Landuse_Intensity_Indicator_Tau-79
Emissions_CO2_Land_Land_use_Change-78
Resources_Peatland_Degraded-3

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
Current develop: `PR611_SSP2-Ref-PeatOn`
This PR: `PR611Final4_SSP2-Ref-PeatV2`

  - Current develop branch default : 24.6 min
  - This PR's default :  24.6 mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [x] content review done (at least 1)
- [x] RSE review done (at least 1)

### CODE REVIEW

File: modules/58_peatland/v2/equations.gms [hunk 1]
```diff
@@ -33,24 +33,17 @@
...
     pc58_peatland(j2,"forestry")
-    + ((vm_land_forestry(j2,"plant")-pcm_land_forestry(j2,"plant"))*p58_scaling_factor(j2));
-
-*' This constraint avoids the conversion of intact peatland into rewetted peatland. 
-*' In each cluster, rewetted peatland area can only increase if no intact peatland area is lost. 
-*' Therefore, rewetted peatland area can only increase if degraded peatland area (`landDrained58`) declines.
-
- q58_peatland_rewet(j2) ..
- v58_expansion(j2,"rewetted") * v58_reduction(j2,"intact") =e= 0;
+    + ((vm_land_forestry(j2,"plant")-pcm_land_forestry(j2,"plant"))*p58_scaling_factor(j2,"forestry"));
```
Comment (tscheypidi) [inline modules/58_peatland/v2/equations.gms:46]: can these three equations now be merged into one? They are now identical for crop, past and forestry, right?

**Review by tscheypidi — CHANGES_REQUESTED** [review]

File: modules/58_peatland/v2/equations.gms (hunk 1 above)
Comment (flohump) [reply modules/58_peatland/v2/equations.gms:46]: Two different interfaces are used in these equations: `vm_land` and `vm_land_forestry`. Therefore, it is not possible to merge these 3 equations into one using a set.

File: modules/58_peatland/v2/equations.gms [hunk 2]
```diff
@@ -33,24 +33,17 @@
  q58_peatland_crop(j2)$(sum(ct, m_year(ct)) > s58_fix_peatland) ..
   v58_peatland(j2,"crop") =e=
     pc58_peatland(j2,"crop")
-    + ((vm_land(j2,"crop")-pcm_land(j2,"crop"))*p58_scaling_factor(j2));
+    + ((vm_land(j2,"crop")-pcm_land(j2,"crop"))*p58_scaling_factor(j2,"crop"));
```
Comment (deleip) [inline modules/58_peatland/v2/equations.gms:36]: The example above still uses the old logic for scaling factors, would be good to update that. 
But thinking about a new example I'm actually wondering whether the new method works in all cases: what if a cell has 100ha, out of which 30ha are cropland, 50ha are peatland, and of those 10ha are peatland used for crops. Then we get `p58_scaling_factor(j2,"crop") = 1/3`. If now in the next timestep cropland would expand to the full area of the cell, we would get `v58_peatland(j2,"crop") = 10 + (100 - 30) * 1/3 = 33.33 ha`. But if the full cell area is cropland, also the full peatland area would need to be used for cropland... Are such cases somehow taken care of?

**Review by deleip — CHANGES_REQUESTED** [review]

File: modules/58_peatland/v2/equations.gms (hunk 2 above)
Comment (flohump) [reply modules/58_peatland/v2/equations.gms]: With the new approach such cases are taken care for.

**Review by deleip — APPROVED** [review]
Looks good to me. Just some smaller comments on readability/code simplification left :)

File: modules/58_peatland/v2/realization.gms [hunk 3]
```diff
@@ -20,19 +20,20 @@
...
 *' Degraded peatland area (crop, past, forestry and unused) depends on managed land. 
 *' Area for peat extraction (peatExtract) is fixed. 
-*' Rewetted peatland area can only increase if degraded peatland area declines and intact peatland area remains constant.
+*' Rewetted and intact peatland have the same GHG emission factors, which avoids that intact is converted to rewetted peatland area.
 *' @stop
 
 
 *'
-*' @limitations Peatland area and GHG emissions are fixed to 2015/2020 levels for the historic period, 
+*' @limitations Peatland area and GHG emissions are fixed to 2020 levels for the historic period,
```
[the remark this answers is not in the document]
Comment (flohump) [reply modules/58_peatland/v2/realization.gms]: yes. changed.

File: modules/58_peatland/v2/equations.gms [hunk 4]
```diff
@@ -7,66 +7,79 @@
...
+q58_scalingFactorExp(j2,manPeat58)$(sum(ct, m_year(ct)) > s58_fix_peatland) ..
+  v58_scalingFactorExp(j2,manPeat58) =e= 
+   (m_peatLandLeft(pc58_peatland,"land58",v58_peatland,pc58_peatland) / m_peatLandLeft(pcm_land,"land",v58_manLand,pc58_manLand))
+   $(m_peatLandLeft(pc58_peatland,"land58",pc58_peatland,pc58_peatland) > 1e-10 AND m_peatLandLeft(pcm_land,"land",pc58_manLand,pc58_manLand) > 1e-10)
+ + 0$(m_peatLandLeft(pc58_peatland,"land58",pc58_peatland,pc58_peatland) <= 1e-10 OR m_peatLandLeft(pcm_land,"land",pc58_manLand,pc58_manLand) <= 1e-10);
 
 *' Costs for peatland degradation and rewetting
 
  q58_peatland_cost_full(j2) ..
   vm_peatland_cost(j2) =e= v58_peatland_cost(j2);
```
[the remark this answers is not in the document]
Comment (flohump) [reply modules/58_peatland/v2/equations.gms]: I removed `v58_peatland_cost` and took care of adjustments in the magpie4 R library.
In this context, I reintroduced `v58_peatland_cost_annuity` with a new set `cost58`, which replaces the 3 individual constraints and variables. This simplifies the code and avoids further adjustments in the magpie4 R library.

File: modules/58_peatland/v2/sets.gms [hunk 5]
```diff
@@ -7,13 +7,16 @@
...
 
-  landDrained58(land58) Peatland land types
+  drained58(land58) Drained peatland categories
     / crop, past, forestry, unused /
 
-  landDrainedUsed58(land58) Peatland land types
+  manPeat58(land58) Drained and managed peatland categories
+    / crop, past, forestry /
+
+  manLand58(land) Managed main land categories
```
[the remark this answers is not in the document]
Comment (flohump) [reply modules/58_peatland/v2/sets.gms]: removed

**Comment by flohump** [comment]
> Looks good to me. Just some smaller comments on readability/code simplification left :)

Thanks for the detailed review!
I added you as co-author of the peatland module. 
After the update of the cost variables, I sucessfully produced new test runs.

File: core/macros.gms [hunk 6]
```diff
@@ -97,3 +97,15 @@ $macro m_carbon_stock(land,carbon_density,item) \
...
             sum((&&sets_sub), land(j2,&&sets_sub) * sum(ct, carbon_density(ct,j2,&&sets_sub,ag_pools)))$(sameas(stockType,"actualNoAcEst"));
+
+* macro for merging managed land area into peatland categories
+$macro m_peatLandMerge(land,landForestry,set) \
+   land(&&set,"crop")$(sameas(manPeat58,"crop")) \
+   + land(&&set,"past")$(sameas(manPeat58,"past")) \
+   + landForestry(&&set,"plant")$(sameas(manPeat58,"forestry"))
+
+* macro for peatland expansion factor
+$macro m_peatLandLeft(pclandFull,setFull,vland,pcland) \
```
[the remark this answers is not in the document]
Comment (flohump) [reply core/macros.gms]: renamed to `m58_LandLeft`

**Pull request merged**